enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -45,43 +66,62 @@ Organizations desire to reuse their existing enterprise log solutions to store c
 The specific goals of this proposal are:
 
 * Provide an API that identifies log sources and their desired forwarding outputs.
-* Send logs to an Elasticsearch cluster not managed by the cluster logging infrastructure
-* Send logs to a fluentd instance not managed by the cluster logging infrastructure via fluentd's `forward` protocol
+* Send logs to endpoints not managed by the cluster logging infrastructure:
+  * An Elasticsearch cluster
+  * An endpoint that accepts the `fluent_forward` protocol (fluentd, fluentbit, others...).
+  * An endpoint that accepts syslog via UDP, TCP or TLS.
+  * Others <!-- XXX(alanconway) what is the target set for GA? -->
+* Provide an extensible API so that will allow new output types to be added in future.
 * Allow deployment of log forwarding without deploying the entirety of the cluster logging infrastructure (e.g. Kibana, Elasticsearch)
 * Support TLS between the collector and destination if so configured
 * Use the same log forwarding logic to write logs to a specified output or to the default storage managed by cluster logging (e.g. Elasticsearch).
+* Deploy global forwarders that can forward logs from any container in the cluster.
```

Comment:
"global forwarding configurations"?

Reply:
In latest draft we have a ClusterLogForwarder, but we will have more than one LogForwarder in future - at least for namespace-scoped LogForwarders.  I will fight for "forwarder" over "forwarding configuration" thought. All CRs are "configurations" and the cool think about k8s is you can use them without knowing how they map to processes, nodes, pods etc. So if I'm a developer and I want to control logs from my namespace, I want "my own forwarder". I don't care if all the forwarders in the cluster are really implemented by a single fluentd daemonset, or a routed AMQP network, or a flock of carrier pigeons as long as "my forwarder" does what I tell it. So from that perspective it is a distinct service/object/thing whatever you want to call it, regardless of how it really works.

Reply:
Another way to look at it: k8s resources are about _ownership_ and _control_. If/when we have multiple forwarders, it's because bob controls his namespace, mary controls heres and the admin controls the cluster. So the "forwarders" are distinct things because the are owned and controlled by distinct roles. That's completely separate from the structure of what they control (even if it sometimes aligns) Just like users have notionally separate home directories - that has nothing to do with how the bytes get laid down on the disk, nor does it matter as long as the OS preserves the illusion that your homedir is "your own space".